### PR TITLE
Implement various features for `estate show/list`

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
This implements the following features:

* Receives the `request_username` and `request_uid` from the client and passes it through to the scripts,
* Allows command outputs to be `sequential`, this disables the interlacing of outputs between nodes. The underlining commands are still ran concurrently and buffered
* Allows a command to be ran against every node in the cluster with `has_context: all_nodes`

The `has_context: all_nodes` is the biggest change in this PR. It required making a distinction between explicit context and implicit context. The commands still have a "pseudo context" when ran against every node, but this is hidden from the client. This means `has_context` is now a bit of a misnomer with the possible meanings:

* Has an explicit context, either a `node` or `group` from the CLI,
* Has no context and runs the default script instead, and
* Somewhere in between. No context was specified on the CLI but the command is still ran against each node with rank lookup.

The following changes have been made to the estate commands:

* The `e'grow` and `e'shrink` commands now display the username and uid in the slack message
* The `e'show` has been implemented as request with a `aws` and `default` script
* The `e'list` uses the new `has_context: all_nodes` mechanism to wrap `e'show`. This allows each node to run a script with a different rank

### CODE REVIEW

File: app/models/command.rb [hunk 1]
```diff
@@ -41,6 +41,15 @@ class Command < Hashie::Dash
   property :confirmation, default: nil
   property :scripts,      default: []
   property :has_context,  default: true
+
+  # Flags the output to be displayed sequentially, the jobs will still run
+  # concurrently
+  property :sequential_output, default: false
```
Comment (benarmston) [inline app/models/command.rb:47]: I can't see any reason for us to have this additional complexity here.  If people want the output to not be interleaved they can either pipe the output through `sort` or event better through `dshbak`.

Am I missing something here?

File: libexec/estate-grow/default.sh [hunk 2]
```diff
@@ -103,7 +103,7 @@ for channel in "${channels[@]}"; do
   "text": "Received a \`flight-estate\` modification request!",
   "attachments": [
     {
-      "text": "*Cluster*: $flight_ESTATE_cluster\n*Action*: $__flight_ESTATE_action\n*Machine Type*: $machine_type\n*Number*: $number"
+      "text": "*Unauthenticated User:* $request_username ($request_uid)\n*Cluster*: $flight_ESTATE_cluster\n*Action*: $__flight_ESTATE_action\n*Machine Type*: $machine_type\n*Number*: $number"
```
Comment (benarmston) [inline libexec/estate-grow/default.sh:106]: I think `User (unauthenticated)` sounds a bit less scary than `Unauthenticated User`.

File: libexec/estate-show/aws.sh [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  --query InstanceType.Value \
+  --instance-id "$ec2_id" \
+  --region "$aws_region"
+)
+
+exit_code=$?
+
+if [ ${exit_code} -eq 0 ] ; then
+  machine_type=$(case "$aws_instance_type" in
+  ('"t2.small"') echo "general-small" ;;
```
Comment (benarmston) [inline libexec/estate-show/aws.sh:24]: What's the reason for only `general-small` having `echo`?

**Review by benarmston — CHANGES_REQUESTED** [review]
@williammccumstie I was expecting this to look much more like https://github.com/openflighthpc/flight-action/commit/7d677f01c7a4fc6803a5f084c32bc79583270c5b.

Perhaps down the line, we'll need an `all nodes` context for commands/actions, but we don't need it yet.  As the client-side special casing implementation is much simpler, we should go with that until.  At least until we need the `all nodes` context.

I've added more comments in line.

File: app.rb [hunk 4]
```diff
@@ -172,7 +172,10 @@ def not_found_relation(rio)
     show
 
     create do |attributes|
-      ticket = Ticket.new(arguments: attributes[:arguments])
+      attrs = [:arguments, :request_uid, :request_username].map do |key|
```
[the remark this answers is not in the document]
Comment (WilliamMcCumstie) [reply app.rb:175]: :+1:

File: app/models/command.rb (hunk 1 above)
Comment (WilliamMcCumstie) [reply app/models/command.rb]: Removed in rebase

File: libexec/estate-show/aws.sh (hunk 3 above)
Comment (WilliamMcCumstie) [reply libexec/estate-show/aws.sh]: Fixed in rebase

Comment (WilliamMcCumstie) [reply libexec/estate-show/aws.sh]: https://github.com/openflighthpc/flight-action-api/pull/16/commits/7ea97b6772369b11df2ce2e789119136ef04e08f

File: libexec/estate-grow/default.sh (hunk 2 above)
Comment (WilliamMcCumstie) [reply libexec/estate-grow/default.sh]: https://github.com/openflighthpc/flight-action-api/pull/16/commits/7ea97b6772369b11df2ce2e789119136ef04e08f

**Pull request merged**